tests/integration/test_reshape.py

```diff
@@ -132,13 +146,91 @@ def test_1d(self):
             np.reshape(self.anp, (200,)),
         )
 
-    def test_ravel(self):
+    @pytest.mark.xfail
+    @pytest.mark.parametrize("order", ("C", "F", "A", "K", None), ids=str)
+    def test_ravel(self, order):
+        # In Numpy, pass with 'K'
+        # In cuNumeric, when order is 'K', raise ValueError:
+        # order 'K' is not permitted for reshaping
         a = num.array(self.anp)
         assert np.array_equal(
-            num.ravel(a),
-            np.ravel(self.anp),
+            num.ravel(a, order=order),
+            np.ravel(self.anp, order=order),
         )
 
+    @pytest.mark.xfail
+    def test_ravel_a_none(self):
+        # In Numpy, pass and returns [None]
+        # In cuNumeric, raises AttributeError:
+        # 'NoneType' object has no attribute 'ravel'
+        assert np.array_equal(
+            num.ravel(None),
+            np.ravel(None),
+        )
+
+
+@pytest.mark.parametrize("shape", (0, (0,), (1, 0), (0, 1, 1)), ids=str)
+def test_reshape_empty_array(shape):
+    a = num.arange(0).reshape(0, 1)
+    anp = np.arange(0).reshape(0, 1)
+    assert np.array_equal(
+        num.reshape(a, shape),
+        np.reshape(anp, shape),
+    )
+
+
+class TestReshapeErrors:
+    def setup_method(self):
+        self.a = num.arange(24)
+        self.shape = (4, 3, 2)
+
+    def test_a_none(self):
+        # In Numpy, it raises ValueError: cannot reshape array
+        msg = "'NoneType' object has no attribute"
+        with pytest.raises(AttributeError, match=msg):
+            num.reshape(None, self.shape)
```

Comment:
Numpy raises a `ValueError` here:
```
In [5]: np.reshape(None, (4,3,2))

ValueError: cannot reshape array of size 1 into shape (4,3,2)
```
so this would be an `xfail`

Reply:
Fixed.